### PR TITLE
Expose DataFile serde functions

### LINKED ISSUE #1562: Expose DataFileSerde as functions
### Is your feature request related to a problem or challenge?

Right now `DataFileSerde` is not usable outside `spec` crate. 

Serde datafiles can actually be useful at some times. In the ongoing DataFusion write support work, write node will write data and generate data files, and it would be ideal to serialize data files and pass them to commit node for the distributed execution.

### Describe the solution you'd like

We don't have to expose `DataFileSerde` directly. Instead, we can expose the serde functions:
```
pub fn serialize_data_file_to_json

pub fn deserialize_json_to_data_file
```

### Willingness to contribute

I can contribute to this feature independently

### PR DESCRIPTION
**Pull request by CTTY**
## Which issue does this PR close?

- Closes #1562 

## What changes are included in this PR?
- Added functions to provide serde for DataFile
- Have `DataFileSerde::try_from` take in `FormatVersion` directly

## Are these changes tested?
Added uts

### CODE REVIEW

File: crates/iceberg/src/spec/manifest/mod.rs [hunk 1]
```diff
@@ -1056,4 +1091,120 @@ mod tests {
         assert!(!partitions[2].clone().contains_null);
         assert_eq!(partitions[2].clone().contains_nan, Some(false));
     }
+
```
Comment (CTTY) [inline crates/iceberg/src/spec/manifest/mod.rs:1095]: As discussed [here](https://github.com/apache/iceberg-rust/pull/1511/files#r2222019640), I'm planning to use snapshot test instead

File: crates/iceberg/src/spec/manifest/mod.rs [hunk 2]
```diff
@@ -1056,4 +1093,155 @@ mod tests {
...
+            "upper_bounds": [],
+            "key_metadata": null,
+            "split_offsets": [],
+            "equality_ids": [],
+            "sort_order_id": null,
+            "first_row_id": null,
+            "referenced_data_file": null,
+            "content_offset": null,
+            "content_size_in_bytes": null
+        });
```
Comment (CTTY) [inline crates/iceberg/src/spec/manifest/mod.rs:1222]: After some tinkering, I think using `serde_json::Value` to check the json output gives the most readable test.

Also I found a tricky corner case: `DataFile`'s serializer won't preserve the order of `DataFile`'s array like `column_sizes`. Right now I just make the array sizes to be one to work around that, maybe there is a better solution?

Comment (liurenjie1024) [reply crates/iceberg/src/spec/manifest/mod.rs:1222]: >Also I found a tricky corner case: DataFile's serializer won't preserve the order of DataFile's array like column_sizes. Right now I just make the array sizes to be one to work around that, maybe there is a better solution?

I guess the randomness comes from `HashMap`, but since now you use `serde_json::Value` to do the check, I guess the order doesn't matter?

Comment (liurenjie1024) [reply crates/iceberg/src/spec/manifest/mod.rs:1222]: Seems it still matters in below outtput. One approach is to change [this function](https://github.com/apache/iceberg-rust/blob/8dd4a2216728c6fcad42a4d48ea7e39bef30517d/crates/iceberg/src/spec/manifest/_serde.rs#L295) to make it determined:
```rust
#[cfg(test)]
// sort the keys.
```
